### DISCUSSION TITLE
Can I use the project to intercept syscalls and modify arguments

### DISCUSSION THREAD

**Discussion by ghost** [bpftrace/bpftrace#2310] [Q&A]
I want to intercept some syscalls and change their arguments to something else, is it possible to do with the bpftrace?

I know seccomp notify can do something similar

**Comment by fbs** [comment]
afaik there are no bpf helpers that allow modifying the arguments, if there are we can consider add support for it.

**Reply by fripSide** [reply]
bpf_probe_write_user can modify the arguments if the pointer content is stored in user space.
